File: src/optic/lua/ssa/instructions/Branch.java
package optic.lua.ssa.instructions;

import optic.lua.ssa.*;
import org.jetbrains.annotations.NotNull;

import java.util.List;
import java.util.stream.Stream;

public final class Branch implements Step {
	private final Register condition;
	private final List<Step> body;

	public Branch(Register condition, List<Step> body) {
		this.condition = condition;
		this.body = body;
	}

	@Override
	public String toString() {
		return "if " + condition;
	}

	@Override
	public @NotNull Stream<Step> children() {
		return body.stream();
	}
}

File: src/optic/lua/ssa/instructions/Branch.kt
package optic.lua.ssa.instructions

import optic.lua.ssa.Register
import optic.lua.ssa.Step
import java.util.stream.Stream

class Branch(val condition: Register, val body: List<Step>) : Step {
    override fun toString(): String {
        return "if $condition"
    }

    override fun children(): Stream<Step> {
        return body.stream()
    }
}
